### ISSUE TITLE
Configurable background refresh

### ISSUE THREAD

**Issue by jr3us** [bennettpeter/android-MythTV-Leanfront#82]
I have copy pasted the emails from the mailing list, and annotated in a couple of places below:

After using the leanfront version on firetv, my wife has found a feature that is needed. :-).  after you have been watching something, and return to the current recordings listings, new recordings  are not shown, and have to be rescanned by going to the tools, and then waiting until that completes before you can see what’s currently available.

After I showed her how to get the list updated… she says.. take me back to the original mythfrontend. :-)

The specific case involved changing over to watch the news that gets scheduled for 12pm, or 5pm locally. If the scan could occur in the ‘background’  to update new stuff since the last scan, that would be a good thing.  :-)

Videos section doesn’t need to be updated as it only changes occasionally.

Regards!

Jay

Leanfront refreshes the list in the background once per hour. The refresh can take up to a minute. I could make it configurable so that you can try to achieve a balance between system load and wife acceptance factor.

The videos section is automatically reloaded as well once per hour.

I could make the video reload interval separate from the recording interval and make both configurable.

-- I like the sound of a separate interval for videos vs recordings! 

-- I have a number of videos in the videos section, and so takes awhile to startup when opening leanfront. Is it possible to also reuse info previously generated for just videos?

I do like the ability to refresh on a configurable interval.

It would not be easy to immediately update when a recording starts or ends like mythfrontend does.

The recording list is updated when the recording starts, so if the recording is still in progress it shows in the list, after the refresh whether manual or automatic.

-- I don't see a negative to it being shown in the list when it is started, but very little in it. 
or am I misunderstanding.

That is what we get in Mythtv-frontend.

Please open an issue in github if you want me to make the change.

Peter

Thanks!!

_______________________________

**Comment by bennettpeter** [comment]
TLDR:

Add a setting to support a user selectable interval between background updates of the program listing. Currently it is hard coded at 60 minutes.

**Comment by bennettpeter** [comment]
Added a setting in Advanced in v0-481-g2f4a197-m1.2.1-2-gc3114cd837. Setting has to be a multiple of 4 minutes, minimum 4 minutes.
